kitsune/wiki/tests/test_views.py

```diff
@@ -684,3 +685,170 @@ def test_it_works(self):
+        * But we have a fallback locale for one of the locales of ACCEPT_LANGUAGE header
+           ** This fallback locale is mentioned in kitsune/settings.py
+        *** Test that the fallback locale is being used
+        """
+
+        # A Client with de, an, ja language choices in ACCEPT_LANGUAGE Header
+        header = 'de,an;q=0.7,ja;q=0.3,'
+        # Create a translated document in es locale version.
+        # And try to get data of de locale vesion of document with above mentioned client
+        doc_content = self.get_data_from_translated_document(header=header,
+                                                             create_doc_locale='es',
+                                                             req_doc_locale='de')
+        # As the document is not translated into de locale version, also there are no locale
+        # in client header which the document is translated. But we have a fallback locale
+        # mentioned in kitsune/settings.py for the an locale. it is es. As the document is
+        # translated into es version, the user should see es version of the document
+        en_content = 'This article is in English'
+        trans_content = 'This article is translated into es'
+        assert en_content not in doc_content
+        assert trans_content in doc_content
+
+    def test_custom_locale_mapping(self):
+        """
+        Test that configured locale fallbacks are used for requested locale
+
+        * The Article is not translated into requested locale and any of the header locale
+        * None of the Header locale have fallback locale
+        * But The requested locale have configured fallback locales and the article is translated
+          into one of the configured fallback locale
+          ** This custom fallback locale is listed in kitsune/wiki/config.py
+        *** Test that the Article is showing in the configured fallback locale
+        """
+
+        # Create a translated document in bn-BD locale and try to get bn-IN version
+        # of the document. While we the client have no language choices.
+        doc_content = self.get_data_from_translated_document(header=None,
+                                                             create_doc_locale='bn-BD',
+                                                             req_doc_locale='bn-IN')
+        # As the Document is not translated into bn-IN and there is no locale of
+        # ACCEPT_LANUGAE header which the document to, But there is a custom wiki fallback
+        # mapping available for bn-IN. the fallback locale is bn-BD. So the user should
+        # see the document in bn-BD.
+        en_content = 'This article is in English'
+        trans_content = 'This article is translated into bn-BD'
+        assert en_content not in doc_content
+        assert trans_content in doc_content
+
+    def test_custom_locale_mapping_for_header_locale(self):
+        """
```

Comment:
Entire section rewrite:
"Test that defined locale fallbacks are used as header locales, with the following conditions:

    the article is not localized into the user-requested locale
    the article is not localized to any of the locales listed in the ACCEPT_LANGUAGE header
    the header locales do not have a defined fallback locale OR the article is not localized into any of those defined fallback locales
    the user-requested locale has a defined fallback locale, but the article is not localized into it
    one of the locales listed in the ACCEPT_LANGUAGE headers has a defined fallback locale and the article is localized into that fallback locale

Test that the defined fallback locale is used"